compositor_integration_tests/src/helpers.rs

```diff
@@ -0,0 +1,173 @@
+            break;
+        }
+    }
+
+    if is_update {
+        println!("Updating output dump: {:?}", file_name.as_ref());
+        let save_path = PathBuf::from(env!("CARGO_MANIFEST_DIR"))
+            .parent()
+            .unwrap()
+            .join("snapshot_tests")
+            .join("snapshots")
+            .join("rtp_packet_dumps")
+            .join("outputs")
+            .join(file_name);
+        std::fs::write(save_path, &output_dump).context("Failed to write output dump")?;
+    }
+
+    Ok(output_dump.freeze())
+}
+
+pub fn compare_dumps(
+    expected: &Bytes,
+    actual: &Bytes,
+    time_range: (Duration, Duration),
+) -> Result<()> {
+    fn unmarshal_packets(data: &Bytes) -> Result<Vec<Packet>> {
+        let mut packets = Vec::new();
+        let mut read_bytes = 0;
+        while read_bytes < data.len() {
+            let packet_size = u16::from_be_bytes([data[read_bytes], data[read_bytes + 1]]) as usize;
+            read_bytes += 2;
+
+            if data.len() < read_bytes + packet_size {
+                break;
+            }
+
+            // TODO(noituri): Goodbye packet
+            let packet = Packet::unmarshal(&mut &data[read_bytes..(read_bytes + packet_size)])?;
+            read_bytes += packet_size;
+
+            packets.push(packet);
+        }
+
+        Ok(packets)
+    }
+
+    fn find_packets_in_range(
+        mut packets: Vec<Packet>,
+        range: (Duration, Duration),
+        clock_rate: u32,
+    ) -> Result<Vec<Packet>> {
+        let start_timestamp = (range.0.as_secs_f32() * clock_rate as f32) as u32;
+        let end_timestamp = (range.1.as_secs_f32() * clock_rate as f32) as u32;
+
+        packets.retain(|p| {
+            let timestamp = p.header.timestamp;
+            timestamp >= start_timestamp && timestamp <= end_timestamp
+        });
+
+        Ok(packets)
+    }
+
+    let expected_packets = unmarshal_packets(expected)?;
+    let actual_packets = unmarshal_packets(actual)?;
+
+    let clock_rate = expected_packets
+        .first()
+        .map(|p| match p.header.payload_type {
+            // Video
+            96 => 90000,
+            // Audio Opus
+            97 => 48000,
+            _ => unreachable!("Unsupported payload type"),
+        })
+        .ok_or_else(|| anyhow::anyhow!("No packets found"))?;
+
+    let expected_packets = find_packets_in_range(expected_packets.clone(), time_range, clock_rate)?;
+    let actual_packets = find_packets_in_range(actual_packets.clone(), time_range, clock_rate)?;
+    for (expected_packet, actual_packet) in expected_packets.iter().zip(actual_packets.iter()) {
+        validate_packets(expected_packet, actual_packet)?;
+    }
+
+    Ok(())
+}
+
+// TODO(noituri): Implement this properly
+fn validate_packets(expected: &Packet, actual: &Packet) -> Result<()> {
+    if expected.header.payload_type != actual.header.payload_type {
+        return Err(anyhow::anyhow!("Payload type mismatch"));
+    }
+
+    Ok(())
+}
```

Comment:
This is not working properly as of now. I need to decode the packets to make it work reasonably. I wanted to include it in this PR but there's already a lot of code so I'll do that in a follow-up PR.